src/main/java/at/hannibal2/skyhanni/features/mining/FlowstateHelper.kt

```diff
@@ -0,0 +1,123 @@
+import at.hannibal2.skyhanni.utils.RegexUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SkyBlockItemModifierUtils.getEnchantments
+import at.hannibal2.skyhanni.utils.inPartialSeconds
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object FlowstateHelper {
+    private val config get() = SkyHanniMod.feature.mining
+    private var timeSinceBreak = SimpleTimeMark.farPast()
+    private var display: List<Renderable>? = null
+
+    private var blockBreakStreak = 0
+
+    private val pickobulusPattern by RepoPattern.pattern(
+        "mining.pickobulus.blockdestroy",
+        "§7Your §r§aPickobulus §r§7destroyed §r§e(?<amount>\\d+) §r§7blocks!"
+    )
+
+    @HandleEvent(onlyOnSkyblock = true)
+    fun onBlockMined(event: OreMinedEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (hasFlowstate() == null) return
+
+        timeSinceBreak = SimpleTimeMark.now().plus(10.seconds)
+        blockBreakStreak += event.extraBlocks.values.sum()
+        createDisplay()
+        attemptClearDisplay()
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (hasFlowstate() == null) return
+
+        pickobulusPattern.matchMatcher(event.message) {
+            timeSinceBreak = SimpleTimeMark.now().plus(10.seconds)
+            blockBreakStreak += group("amount").toInt()
+            createDisplay()
+            attemptClearDisplay()
+        }
+    }
+
+    private fun attemptClearDisplay() {
+        DelayedRun.runDelayed(10.seconds) {
+            if (timeSinceBreak.isInFuture()) return@runDelayed
+            blockBreakStreak = 0
+            createDisplay()
+        }
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (!config.flowstateHelper) return
+        if (hasFlowstate() == null) return
+
+        if (display == null || timeSinceBreak.isInFuture()) createDisplay()
+        display?.let {
+            config.flowstateHelperPosition.renderRenderables(
+                listOf(Renderable.verticalContainer(it, 1)),
+                posLabel = "Flowstate Helper",
+                addToGuiManager = true
+            )
+        }
+    }
+
+    private fun createDisplay() {
+        val helperDisplay: MutableList<Renderable> = mutableListOf()
+
+        helperDisplay.add(Renderable.string("§d§lFlowstate Helper"))
+        helperDisplay.add(createDisplayTimer())
+        helperDisplay.add(createDisplayBlock())
+        helperDisplay.add(Renderable.string("§6+${getSpeedBonus()}⸕"))
```

Comment:
You can cache all values until block break (except the timer). The list creation it self is not that expensive.